api/api_test.go

```diff
@@ -29,3 +37,94 @@ func testHTTPServer(
 
 	return config, ln
 }
+
+// nextPort is the next port to use for the API server.
+var nextPort int32 = 28200
+
+// restVaultServer runs an instance of the Vault server in development mode.
+// This requires that the vault binary is installed and in the $PATH.
+func testVaultServer(t *testing.T) (*Client, func()) {
```

Comment:
Is there a reason to invoke the binary rather than simply launching three test cores with listeners using TestCluster? There are a number of benefits -- it includes HA, fully-set-up clients using TLS, and more.